script/tool/lib/src/publish_plugin_command.dart

```diff
@@ -135,6 +269,14 @@ class PublishPluginCommand extends PluginCommand {
       @required String remoteUrl,
       @required bool shouldPushTag}) async {
     final String tag = _getTag(packageDir);
+    if (argResults[_dryRunFlag] as bool) {
+      _print('DRY RUN: Tagging release $tag...');
+      if (!shouldPushTag) {
+        return;
+      }
+      _print('DRY RUN: Pushing tag to $remote...');
+      return;
+    }
```

Comment:
Duplicating all the logic and log messages for all dry run commands will be a maintenance mess. It also means a dry run is only testing the special dry-run logic, not the actual logic that will run, which is a landmine waiting to happen.

I'd much rather we have a big "===== DRY RUN =====" (or similar) banner at the start of the output, and have the dry-run short-circuit individual commands, but otherwise run through the normal logic.

Reply:
done